src/provider/handler/mappers/postMapper.js

```diff
@@ -1,36 +1,50 @@
-export function mapPost(post) {
-  let result = {
-    type: {
-      value: 'article'
-    },
-    media_group: [],
-    extensions: {},
-    content: {},
-    link: {}
-  };
-
-  result.id = post.id;
-  result.title = (post.title && post.title.rendered)?post.title.rendered:'';
-  result.publish = post.date;
+export function mapPost(mediaItems) {
+  return function(post) {
+    const { id, date: publish, link, title } = post;
 
-  //the post's link that will be used when a user shares the post
-  result.link = {
-    type: 'text/html',
-    rel: 'alternate',
-    href: post.link
-  };
+    //if we can find the post's media id then let's add its url to our media_group
+    const media_group = mediaItems.reduce((result, mediaItem) => {
```

Comment:
2 thinks here : 
* you can filter the mediaItem array before map / reduce
* you're transforming an array to an array here, so map is more appropriate

the way you've written it, the final array will have only one item, cause you are disregarding the result when returning. reduce doesn't automatically append the returned value, it acts as an accumulator

Reply:
yes, the point here is to always return an array with 1 item (or empty) - the reduce is just used to loop through the array and find a match for the post id (if exists). only one match is possible.

Reply:
Ok, I see.
But I still think that filter / map is more expressive here.
Your logic is : 
- find the item (you can even use array.find to make sure you get only one match)
- transform (with map)

so using both methods is better - especially if you pick find cause then you don't need to iterate over the array once the match is found

Reply:
ok, used find (no need to map here)

Reply:
find expects a predicate, not a transformation : https://www.w3schools.com/jsref/jsref_find.asp

it will return you one item. so indeed, you don't need to map after that, but that will return you the original item. you need to apply the transformation after that